app/src/main/java/com/mpieterse/stride/workers/PushWorker.kt

```
@@ -0,0 +1,69 @@
+
+        // push habit creates first
+        anyApplied = habits.pushHabitCreates() || anyApplied
+        anyApplied = checkins.pushBatch() || anyApplied
+
+        if (anyApplied) PullWorker.enqueueOnce(applicationContext)
+        Result.success()
+    } catch (e: HttpException) {
+        // surface HTTP details to Debug UI
+        val data = workDataOf("error" to "HTTP ${e.code()} ${e.message()}")
+        Result.failure(data)
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)